content/en/docs/ops/common-problems/security-issues/index.md

```diff
@@ -93,9 +93,13 @@ matching requests should flow through. If all requests continue to be denied, yo
 1. Make sure that your authorization policies with ALLOW action don't use any HTTP only fields for TCP traffic.
 Otherwise, Istio ignores the ALLOW policies as if they don't exist.
 
+1. A HTTP Response with the value `RBAC: Access Denied` indicates an authorization policy is in effect. You can determine the authorization policy in effect by running `istioctl x authz check POD-NAME`.
+
 1. Make sure that your authorization policies with DENY action don't use any HTTP only fields for TCP traffic.
 Otherwise, Istio ignores the rules with HTTP only fields within the DENY policies as if they don't exist.
 
+1. A HTTP Response with the value `upstream connect error or disconnect/reset before headers. reset reason: connection termination` indicates an authorization policy with HTTP only fields applied to TCP traffic. Read the [port selection documentation](docs/ops/configuration/traffic-management/protocol-selection/) for how Istio determines whether a service using http or tcp protocol.
```

Comment:
I am not sure this is accurate. This error comes from quite a few issues, including mTLS ones, etc. We could just soften the language to suggest this is one possible reason?

Reply:
+1 there are multiple reasons behind this and it's actually more common to happen with mTLS.

And I feel this is a bit confusing, `authorization policy with HTTP only fields applied to TCP traffic` seems indicating it's TCP traffic which seems conflicting with the HTTP response?

Reply:
I'll amend it to suggest that it can be caused by other issues.


I spent a big chunk of a day figuring out why i was getting that error.

```
 root  DESKTOP-7JUP8RO  mnt  c  …  Git  ccoe-landing-zone  k8s   master  119✎  1+  #    k get svc atlantis -o yaml
apiVersion: v1
kind: Service
metadata:
  annotations:
    meta.helm.sh/release-name: atlantis
    meta.helm.sh/release-namespace: dev
  creationTimestamp: "2020-09-07T09:09:44Z"
  labels:
    app: atlantis
    app.kubernetes.io/managed-by: Helm
    chart: atlantis-3.12.2
    heritage: Helm
    release: atlantis
  name: atlantis
  namespace: dev
spec:
  clusterIP: 10.201.62.177
  ports:
  - name: atlantis # renamed it to http-atlantis for Istio to treat it as http protocol after i stumbled across the doc i'm linking in this PR
    port: 80
    protocol: TCP
    targetPort: 4141
  selector:
    app: atlantis
    release: atlantis
  sessionAffinity: None
  type: ClusterIP
root  DESKTOP-7JUP8RO  mnt  c  …  Git  atlantis-reference  k8s   master  2✎  6+  SIGINT  #    k get authorizationpolicy atlantis-oauth -o yaml
apiVersion: security.istio.io/v1beta1
kind: AuthorizationPolicy
metadata:
  annotations:
  creationTimestamp: "2020-09-11T10:29:18Z"
  generation: 7
  name: atlantis-oauth
  namespace: dev
  resourceVersion: "40924986"
  selfLink: /apis/security.istio.io/v1beta1/namespaces/dev/authorizationpolicies/atlantis-oauth
  uid: 4cfc336b-7953-456a-a636-438a2f43b378
spec:
  action: ALLOW
  rules:
  - from:
    - source:
        requestPrincipals:
        - '*'
    to:
    - operation:
        hosts:
        - atlantis.rack.gold
        methods:
        - '*'
    when:
    - key: request.auth.claims["https://borg.dev/github-orgs"]
      values:
      - borg-land
  selector:
    matchLabels:
      app: atlantis
```

Also, is that a correct way of checking custom claims on a jwt? That is broken. I want to update the docs to mention it if it not correct.

Reply:
you shouldn't include `"` in the custom claim unless your claim itself includes the `"` which is a bit unusual. Just `- key: request.auth.claims[https://borg.dev/github-orgs]`

Reply:
thanks for that, it works now. I added a note about not including quotes.

Reply:
One more thing, I think the info here about excluding JWT from certain paths and unauthenticated identity is incorrect.

https://istio.io/latest/docs/reference/config/security/request_authentication/
https://istio.io/latest/docs/reference/config/security/authorization-policy/#Rule

I have an authz that look likes this.

```
apiVersion: security.istio.io/v1beta1
kind: RequestAuthentication
metadata:
  labels:
    app: atlantis
  name: atlantis
  namespace: dev
spec:
  jwtRules:
    - issuer: "https://auth.borg.dev/"
      jwksUri: "https://auth.borg.dev/.well-known/jwks.json"
  selector:
    matchLabels:
      # Match labels should not select the oauth2-proxy, just
      # the application being secured.
      app: atlantis
---
apiVersion: security.istio.io/v1beta1
kind: AuthorizationPolicy
metadata:
  name: atlantis-oauth
  namespace: dev
spec:
  selector:
    matchLabels:
      app: atlantis
  action: ALLOW
  rules:
   - from:
     - source:
         requestPrincipals: ["*"]
     to:
      - operation:
          methods: ["*"]
          hosts: ["atlantis.REDACTED.gold"]
     when:
     - key: request.auth.claims[https://borg.dev/github-orgs]
       values: 
       - "borg-land"
   - from:
       - source:
           requestPrincipals: ["*"]
     to:
       - operation:
           methods: ["*"]
           hosts: ["atlantis.api.REDACTED.gold"]
     when:
       - key: source.ip
         values: ["192.30.252.0/22","185.199.108.0/22","140.82.112.0/20","X.X.X.X/32"]
```

1. I added `requestPrincipals: ["*"]` as explained in the first article about excluding JWTs from /healthz path. I'm still getting RBAC: Access denied.
2. I removed the `from` block from the second rule as mentioned in the second article and that still returned RBAC: Access denied.

It is confusing 😕

Reply:
For the first article, you missed the `-` in the yaml:
```yaml
 - from:
   - source:
       requestPrincipals: ["*"]
 - to:
   - operation:
       paths: ["/healthz"]
```

And it's not of using `requestPrincipals: ["*"]` for excluding, it's the rule `paths: ["/healthz"]` that doesn't have `from` section. Note the `-` prefix in yaml, it means 2 rules in the array, not 1.


The `requestPrincipals: ["*"]` is doing exactly the opposite thing, it requires the JWT to exist instead of excluding it.